### PR TITLE
[DNM] *: add http framework.

### PR DESCRIPTION
**Pull request by siddontang**
This is the first stop fo #562, just introducing a HTTP framework.

@ngaut @BusyJay @disksing @tiancaiamao

### CODE REVIEW

**Comment by siddontang** [comment]
See http://seanmonstar.com/post/141495445652/async-hyper for how to use async hyper.

**Comment by siddontang** [comment]
I think we should link coveralls.io url in readme too.

File: src/server/http_server.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                                     // The connection is closed, no need to call ready again.
+                                     error!("send response failed {:?}", e);
+                                     return;
+                                 }
+
+                                 if let Err(e) = ctrl.ready(Next::write()) {
+                                     error!("set response writable to WRITE failed {:?}", e);
+                                 }
+                             });
+                Next::wait()
```
Comment (BusyJay) [inline src/server/http_server.rs]: Use `Next::timeout` maybe better.

Comment (siddontang) [inline src/server/http_server.rs]: server has handle connection timeout inner directly.

File: src/server/http.rs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+use kvproto::msgpb;
+
+use super::{Result, Error};
+
+pub type OnResponseResult = Result<Option<msgpb::Message>>;
+pub type OnResponse = Box<FnBox(OnResponseResult) + Send>;
+
+pub struct Body {
+    pub pos: usize,
```
Comment (BusyJay) [inline src/server/http.rs]: Why pub?

File: src/server/http.rs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        self.pos = 0;
+        self.data.resize(size, 0);
+    }
+
+    pub fn len(&self) -> usize {
+        self.data.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.data.len() == 0
```
Comment (BusyJay) [inline src/server/http.rs]: `self.data.is_empty()`

Comment (BusyJay) [inline src/server/http.rs]: I think `is_empty` here should mean if `remaining == 0`.

Comment (siddontang) [inline src/server/http.rs]: yep

File: src/server/http.rs (hunk 2 above)
Comment (siddontang) [inline src/server/http.rs]: no need

File: src/server/http_transport.rs [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+    raft_router: Arc<RwLock<R>>,
+    store_peeker: A,
+
+    pub tx: Sender<Task>,
+    rx: Receiver<Task>,
+
+    store_pendings: HashMap<u64, StorePending>,
+}
+
+impl<A, R> Worker<A, R>
```
Comment (BusyJay) [inline src/server/http_transport.rs]: Why not reuse `util::worker::Worker`?

Comment (siddontang) [inline src/server/http_transport.rs]: util worker can only send task in one thread, but I need to support in multi threads.

If you support this feature later, I will change it.

**Comment by siddontang** [comment]
hyper is not stable and has low performance, we should wait now.

**Pull request closed without merging**